Factorial.py

```diff
@@ -0,0 +1,27 @@
+_FAC_TABLE = [1, 1]
+def factorial(n):
+    if n < len(_FAC_TABLE):
+        return _FAC_TABLE[n]
+
+    last = len(_FAC_TABLE) - 1
+    total = _FAC_TABLE[last]
+    for i in range(last + 1, n + 1):
+        total *= i
+        _FAC_TABLE.append(total)
+
+    return total
+
+def rFactorial(n):
+   if n <1:
+       return 1
+   else:
+       return n * factorial( n - 1 )
+
+def main():
+    n = int(input("Enter your number:"))
+    print(factorial(n))
```

Comment:
Would be nice if which algorithm is running/has ran is printed

Reply:
Corrected, if I understand you correctly